### DISCUSSION TITLE
[Projects beta] Allow order of fields to be customized in the Project Board drop down in an issue

### DISCUSSION THREAD

**Discussion by markgammon** [community/community#10044] [Projects and Issues] [answered]
We'd like to be able to set the order of the view for “fields” when they appear in the drop-downs for the project views from the issue (possibly in the “settings” for the project board).  Currently the default/only order is by the creation date of the fields in the board (most recently created fields are at the bottom of the drop down)
Suggest setting this order from within the Project Board -> Settings 
This setting would also set the order in the draft issue view that pops up on the right side when opening a draft issue
![Issue view](https://user-images.githubusercontent.com/92403768/149196234-62ba9cfd-bac2-4160-8d6b-4a196d29236a.png)

**Comment by steebchen** [comment]
This would be pretty cool indeed

**Comment by KristinaKlimashevskaia** [comment]
This will be super helpful

**Comment by ushanlab** [comment]
We really need it!

**Comment by hianhianhian** [comment]
Would be great to have this!

**Comment by jmv74211** [comment]
I was just looking for this feature, it would be very useful 🙏🏼 .

**Comment by Jay-o-Way** [comment]
One year later... Please!

**Comment by sami-akhtar** [comment]
This is a challenge for us. Can you please prioritize this feature as it's pretty important for any team using the project views.

**Comment by alexfornuto** [comment]
DOOO EEEET!

**Comment by Z-K-O** [comment]
Is this being worked on?

Is there currently a workaround?

**Comment by tayloramurphy** [comment]
Looks like there's another discussion here as well that's similar. This would be a very useful feature for us. https://github.com/orgs/community/discussions/13348 My comment in that discussion has more context as well https://github.com/orgs/community/discussions/13348#discussioncomment-4976100

**Comment by raulk** [comment]
The inability to reorder fields makes GitHub Projects super messy for us 😢  This isn't much of an issue for some workflows (e.g. adding an issue to a project board, where the board can be configured with the desired order of columns), but it is problematic for others (e.g. adding an issue to a project from the issue page itself), where the fields appear in the project card in their definition order.

**Comment by darrenmudd** [comment]
For the love of God, why isn't this possible out-of-the-box?

**Comment by ssbarnea** [comment]
The fact that we cannot sort the fields, combined with being forced to have status as first and the fact that project sidebar section is collapsed by default reduces the usability of the project feature to a level that might be counted as "not practical to use".

**Comment by Shrinks99** [comment]
Adding an update for a situation where this would be quite useful:

![Screenshot 2023-06-07 230933](https://github.com/community/community/assets/5672810/5450ea83-aff4-458e-bbb2-c1eea966c824)

In this example I made the error of making the `Feature Target Date` field before the `Feature Start Date` field.  In order to fix this, I'm going to have to delete `Feature Target Date` and remake it in order to reorder the fields so that it appears after `Feature Start Date` to reduce the amount of incorrect inputs.  This would be wildly unfortunate if I didn't realize it early on and had many other fields after that!  Thankfully I only have 4 items to fix!

**Comment by thockin** [comment]
This is the first thing I really got frustrated with!

**Comment by gregmagolan** [comment]
+1

**Comment by rileybroughten** [comment, marked as answer]
You can now reorder fields from the settings page, which will reflect in the item side panel and from the issues page in the "Projects" section. Click to drag and drop the fields in the list. 

Check out the [changelog announcement](https://github.blog/changelog/2023-07-20-github-issues-projects-july-20th-update/#%F0%9F%A4%B8-reorder-fields-in-settings) for more details!

**Reply by tayloramurphy** [reply]
This is amazing to see @rileybroughten! It seems like such a small issue but it's a huge quality of life improvement. Congrats to the team for shipping 😄

**Reply by dequejenn** [reply]
@rileybroughten - this no longer works as of a few weeks ago. I don't know where to open a bug ticket. I used to be able to drag and drop to reorder the customer fields in a project but as of maybe last week or the week before it stopped working.

**Reply by rileybroughten** [reply]
@dequejenn check out the changelog [here](https://github.blog/changelog/2025-07-17-duplicate-issues-create-from-anywhere-and-more/) - you can reorder fields from the `...` menu in projects settings!

<img width="1438" height="353" alt="Screenshot 2025-07-18 at 10 49 03 AM" src="https://github.com/user-attachments/assets/1fdbe103-cbbc-4e51-b666-499c3b7c52b0" />
